pkg/generate/app/pipeline.go

```diff
@@ -456,6 +468,97 @@ func NewAcceptUnique(typer runtime.ObjectTyper) Acceptor {
 	}
 }
 
+type acceptNonExistentImageStream struct {
+	typer     runtime.ObjectTyper
+	getter    imageclient.ImageInterface
+	namespace string
+}
+
+// Accept accepts any non-ImageStream object or an ImageStream that does
+// not exist in the api server
+func (a *acceptNonExistentImageStream) Accept(from interface{}) bool {
+	obj, _, err := objectMetaData(from)
+	if err != nil {
+		return false
+	}
+	gvk, _, err := a.typer.ObjectKinds(obj)
+	if err != nil {
+		return false
+	}
+	gk := gvk[0].GroupKind()
+	if !image.IsKindOrLegacy("ImageStream", gk) {
+		return true
+	}
+	is, ok := from.(*image.ImageStream)
+	if !ok {
+		glog.V(4).Infof("type cast to image stream %#v not right for an unaticipated reason", from)
```

Comment:
shouldn't this return?

also typo in unanticipated.

Reply:
yeah I waffled on this a bit - original thought was to try the get regardless ... but yeah, return true just like `image.IsKindOrLegacy("ImageStream", gk)` makes sense ... part of next push